cmd/nodedaemon/main.go

```diff
@@ -0,0 +1,152 @@
+package nodedaemon
+
+import (
+	"crypto/tls"
+	"flag"
+	nodedaemon "openshift/ksan-operator/internal/node-daemon"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/filters"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+func NewCmd(scheme *runtime.Scheme) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "nodedaemon",
+		Short:         "Commands for running ksan storage node daemon",
+		Long:          `Node daemon managing nodes at ksan storage cluster`,
+		SilenceErrors: false,
+		SilenceUsage:  true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(cmd, scheme, args)
+		},
+	}
+
+	return cmd
+}
+
+func run(_ *cobra.Command, scheme *runtime.Scheme, _ []string) error {
+	setupLog := ctrl.Log.WithName("setup")
+
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	var secureMetrics bool
+	var enableHTTP2 bool
+	var tlsOpts []func(*tls.Config)
+	flag.StringVar(&metricsAddr, "metrics-bind-address", "0", "The address the metrics endpoint binds to. "+
+		"Use :8443 for HTTPS or :8080 for HTTP, or leave as 0 to disable the metrics service.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&secureMetrics, "metrics-secure", true,
+		"If set, the metrics endpoint is served securely via HTTPS. Use --metrics-secure=false to use HTTP instead.")
+	flag.BoolVar(&enableHTTP2, "enable-http2", false,
+		"If set, HTTP/2 will be enabled for the metrics and webhook servers")
+	opts := zap.Options{
+		Development: true,
```

Comment:
IIUC, this is used to set the logger to panic instead of logging an error for panic level logs. Is this something we plan to keep around or is this for development?